### PR TITLE
Add optional sentry release

### PR DESCRIPTION
**Pull request by alexberndt**
investigate option to create a Sentry release within a Dagster repository.

Seems to [work](https://overstory.sentry.io/releases/utils-pipelines%40fa86664e0c39c943579e88977d05b77ae92b6f6d/?project=4507129965969408)

### Notable decisions

- we are using the (docker) `image_name` input as the Sentry project. There is an implicit assumption that the sentry project slug and image names will always remain the same. This is to avoid adding another input to the workflow.
- The Sentry org name is specified as an input in `docker_build_push.yml`. We could also consider this as an org secret and not use a default.

### Next steps

Releases are being created with `<project_name>@<git_commit>`, but we ideally want `<project_name>@<release_semver>`. I want to test if Sentry identifies the version when a release is created (and not on a regular commit). 

Screenshot 2024-05-30 at 13 47 05

### CODE REVIEW

**Comment by alexberndt** [comment]
[Explicit release set in Sentry SDK init](https://trello.com/c/q0YqILsO/55-explicit-release-set-in-sentry-sdk-init)

File: .github/workflows/docker_build_push.yml [hunk 1]
```diff
@@ -222,3 +235,15 @@ jobs:
         name: Add PR-specific tags
         run: |
           gcloud artifacts docker tags add '${{ env.IMAGE_NAME }}:${{ github.sha }}' '${{ format('{0}:pr-{1}', env.IMAGE_NAME, github.event.number) }}'
+
+      - name: Create Sentry release
+        if: ${{ inputs.create_sentry_release }}
+        uses: getsentry/action-release@v1.4.1
```
Comment (chanange) [inline .github/workflows/docker_build_push.yml:241]: Bump to [latest version 1.7.0](https://github.com/getsentry/action-release/releases/tag/v1.7.0) which uses sentry cli 2.24.1 - [2.0 of the cli](https://github.com/getsentry/sentry-cli/releases/tag/2.0.0) has been out for some time and looks like there's no issues with adopting it
```suggestion
        uses: getsentry/action-release@v1.7.0
```

**Review by chanange — APPROVED** [review]

File: .github/workflows/dagster.yml [hunk 2]
```diff
@@ -58,6 +58,11 @@ on:
         default: "1.5.12"
         required: false
         # Chart values https://github.com/dagster-io/dagster/blob/1.5.12/helm/dagster/values.yaml
+      enable_sentry_release_creation:
```
Comment (chanange) [inline .github/workflows/dagster.yml:61]: Keep naming consistent?
```suggestion
      create_sentry_release:
```

File: .github/workflows/dagster.yml [hunk 3]
```diff
@@ -83,10 +91,12 @@ jobs:
       test_dagster: true
       docker_buildx_driver: ${{ inputs.docker_buildx_driver }}
       clear_space: ${{ inputs.clear_space  }}
+      create_sentry_release: ${{ inputs.enable_sentry_release_creation && github.event_name == 'release' }} # Create sentry release on prod release only
```
Comment (chanange) [inline .github/workflows/dagster.yml:94]: we currently don't use the release event when we use this workflow, you'll need to check for whether the github ref is a tag instead

Comment (alexberndt) [reply .github/workflows/dagster.yml:94]: Good catch! Replace with `startsWith(github.ref, 'refs/tags/')`

**Pull request merged**